STYLE_GUIDE.md

```diff
@@ -154,6 +154,62 @@ Prefer:
 
 ## JavaScript
 
+### Variable naming
+
+In order to allow Kedro-Viz to be worked on by many developers from different backgrounds, we try to follow a consistent set of naming conventions. Beyond consistency of form, element names must be easily understood and must convey the function of each element.
+
+✅ DO choose easily readable identifier names.
+
+For example, a variable named `horizontalAlignment` is more English-readable than `alignmentHorizontal`.
+
+✅ DO favour readability over brevity.
+
+The variable name `canScrollHorizontally` is better than `scrollableX`.
+
+✅ DO use verbs in the imperative mood for function/method names.
+
+The function `getSidebarLength()` is better than `sidebarLength()`, as it indicates how it should be used.
+
+✅ DO prefix Boolean variable names to indicate their type, and prefer positive names.
+
+This helps the reader easily infer that this variable is a Boolean value, and helps avoid complicated double-negatives.
+
+Avoid:
+
+```js
+let person = true;
+let age = true;
+let dance = true;
+let notDancing = true;
+```
+
+Prefer:
+
+```js
+let isPerson = true;
+let hasAge = true;
+let canDance = true;
+let isDancing = false;
+```
+
+✅ DO use camelCase for most variable names.
+
+✅ DO use PascalCase for class names.
+
+✅ DO use UPPER_CASE for constant values.
+
+✅ DO use camel-case for file names.
+
+❌ DO NOT use snake_case, except in JSON APIs.
+
+❌ DO NOT use single letter variable names.
+
+There are a few exceptions to this rule as they can be useful when working with D3, or for some common function arguments (e.g. `i` for index). But you should try to use more informative names for the most part.
+
+❌ DO NOT use abbreviations, acronyms or contractions as part of identifier names.
+
+For example, use `getObject` rather than `getObj`. The exception to this is if the acronym is well known for this domain - e.g. JSON.
```

Comment:
This just reminded me that we just adopted `newparams` as our flag name for our parameters PR - this really brings up the point as to whether we should revisit that flag naming?

Reply:
If anything that makes me think I should soften this to say 'avoid' rather than 'never' - I think there are some times where it makes sense and improves legibility.